### ISSUE TITLE
[java] GuardLogStatement reports violation for Log4j parameterized logs

### ISSUE THREAD

**Issue by ThorodanBrom** [pmd/pmd#4703]
**Affects PMD Version:** 6.53.0, 6.55.0

**Rule: GuardLogStatement**
- https://docs.pmd-code.org/pmd-doc-6.53.0/pmd_rules_java_bestpractices.html#guardlogstatement
- https://docs.pmd-code.org/pmd-doc-6.55.0/pmd_rules_java_bestpractices.html#guardlogstatement

**Description:**

PMD reports a violation when [Log4j parameterized logs](https://logging.apache.org/log4j/2.x/javadoc/log4j-api/org/apache/logging/log4j/Logger.html#info-java.lang.String-java.lang.Object-java.lang.Object-java.lang.Object-java.lang.Object-java.lang.Object-java.lang.Object-java.lang.Object-java.lang.Object-java.lang.Object-java.lang.Object-) are used. We are in the process of converting all logs from concatenated strings to parameterized logs, using PMD to find offending logs. But PMD reports the same logs even after updating.

**Code Sample demonstrating the issue:**

```java
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

class Test {
    private static final Logger LOGGER = LogManager.getLogger(Test.class);

    public static void main() {
        LOGGER.info("Info : {} : Request received", LOGGER.getName());
        String s = "Hello World";
        LOGGER.debug("Logging string {}", s);
    }
}

```

**Expected outcome:**
- Both logs do not use concatenation and make use of parameters as specified in the [GuardLogStatement examples](https://docs.pmd-code.org/pmd-doc-6.55.0/pmd_rules_java_bestpractices.html#guardlogstatement)

PMD reports a violation at lines 8 and 10, but that's wrong. That's a false positive.

**Running PMD through:** *Maven*

----

P.S. I saw someone else report the same false-positive on another issue - https://github.com/pmd/pmd/issues/3144#issuecomment-1632847686

**Comment by jsotuyod** [comment]
Under PMD 7.0.0 the second line will not be reported.

The first one, however, will.

```java
LOGGER.info("Info : {} : Request received", LOGGER.getName());
```

this line will always perform a call to `LOGGER.getName()` regardless of the logging level. As any method calls done here can be potentially expensive, they still get flagged. The correct approach would be to use a `Callable` here through a lambda such as:

```java
LOGGER.info("Info : {} : Request received", () -> LOGGER.getName());
```
